### PR TITLE
update with-sentry example

### PR DESCRIPTION
**Pull request by Tomekmularczyk**
Changes:

* moved the configuration from HOC to `_app.js`
* fixed the example, as `componentDidCatch` catches errors during rendering phase, but not within event handlers.

### CODE REVIEW

File: examples/with-sentry/pages/_app.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+    Raven.config(
+      SENTRY_DSN
+    ).install()
+  }
+
+  componentDidCatch (error, errorInfo) {
+    this.setState({ error })
+    Raven.captureException(error, { extra: errorInfo })
```
Comment (timneutkens) [inline examples/with-sentry/pages/_app.js]: This should have `super.componentDidCatch(error, errorInfo)` and the `setState` should be removed. Check the with-componentdidcatch example 👍

Comment (Tomekmularczyk) [reply examples/with-sentry/pages/_app.js]: Got it. I guess `getInitialProps` won't be needed here as well.

**Comment by avdeev** [comment]
rename SENTRY_DSN to SENTRY_PUBLIC_DSN, plz

**Review by timneutkens — APPROVED** [review]

**Comment by Enalmada** [comment]
FYI @Tomekmularczyk @avdeev The [sentry docs](https://docs.sentry.io/platforms/javascript/?platform=javascript) example look different than the example.  I assume raven has turned into @sentry/browser but please set me know if there is some reason next.js needs to use raven instead.

**Comment by timneutkens** [comment]
@Enalmada I merged this yesterday: https://github.com/zeit/next.js/pull/5253

**Pull request merged**